src/core/folder.h

```diff
@@ -179,6 +183,8 @@ private Q_SLOTS:
     bool defer_content_test : 1;
 
     static std::unordered_map<FilePath, std::weak_ptr<Folder>, FilePathHash> cache_;
+    static FilePath cutFilesDirPath_;
+    static std::shared_ptr<const HashSet> cutFilesHashSet_;
```

Comment:
With the help of `FolderModel::setCutFiles` (both of them), I succeeded in reproducing it without desktop too. I think I've found a solution but let's continue our discussion tomorrow -- being too busy, I didn't have enough focus today.